History/HistoryManager/src/main/java/eu/neclab/ngsildbroker/historymanager/repository/HistoryDAO.java

```diff
@@ -41,7 +42,48 @@ public void entityExists(String entityId, String tenantId) throws ResponseExcept
 
 		}
 	}
-	
+
+	public void attributeExists(String entityId, String tenantId, String resolvedAttrId, String instanceId)
+			throws ResponseException {
+		List<String> result;
+		List<String> instanceattrresult;
+		if (tenantId == AppConstants.INTERNAL_NULL_KEY) {
+			result = getJDBCTemplate(null).queryForList(
+					"select attributeid from temporalentityattrinstance WHERE temporalentity_id='" + entityId + "'",
+					String.class);
+			if (!result.contains(resolvedAttrId)) {
+				throw new ResponseException(ErrorType.NotFound, resolvedAttrId + " not found");
+			} else {
+				if (instanceId != null) {
+					instanceattrresult = getJDBCTemplate(null)
+							.queryForList("select attributeid from temporalentityattrinstance WHERE temporalentity_id='"
+									+ entityId + "' and instanceid='" + instanceId + "'", String.class);
+					if (!instanceattrresult.contains(resolvedAttrId)) {
```

Comment:
[Major] This condition is not correct because If In case of attribute id is not exist and instance id exist then it will be return "Instance id is not exist". So please correct it.